app/main.py

```diff
@@ -32,13 +47,13 @@ def main() -> None:
     ]
 
     # run the programs
-    service.run_program(wake_up_program)
-    service.run_program(sleep_program)
+    await run_parallel(service.run_program(wake_up_program))
```

Comment:
You should combine run_sequence and run_parallel and take into account logical order